src/ess/reduce/nexus/workflow.py

```diff
@@ -552,34 +468,35 @@ def _add_variances(da: sc.DataArray) -> sc.DataArray:
 definitions["NXmonitor"] = _StrippedMonitor
 
 
-_common_providers = (gravity_vector_neg_y, file_path_to_file_spec, all_pulses)
+_common_providers = (
+    gravity_vector_neg_y,
+    file_path_to_file_spec,
+    all_pulses,
+    component_spec_by_name,
+    unique_component_spec,  # after component_spec_by_name, partially overrides
```

Comment:
Does the comment mean that we are relying on the behaviour of the pipeline; that it applies the providers in the order that they are given. Can we assume that the behaviour will never change?

Reply:
There are explicit tests for this, e.g., here: https://github.com/scipp/sciline/blob/a90b73c50d4cef1fc8adb231dc2208cf99ef2f0a/tests/pipeline_test.py#L916. That is, this is intentional behavior and not likely to change. If it would, quite a few other workflows would also break, I think.